internal/evaluator/conftest_evaluator.go

```diff
@@ -483,6 +483,37 @@ func createConfigJSON(ctx context.Context, dataDir string, p policy.Policy) erro
 	return nil
 }
 
+// addRuleData creates the $dataDir/rule_data__configuration__/ruledata.json
+// files with the provided custom rule data for each source in the data
+// directory
+func addRuleData(ctx context.Context, dataDir string, p policy.Policy) error {
+	if p == nil {
+		return nil
+	}
+
+	ruleDir := path.Join(dataDir, "rule_data")
+
+	fs := utils.FS(ctx)
+	if err := fs.MkdirAll(ruleDir, 0755); err != nil {
+		return err
+	}
+
+	for i, source := range p.Spec().Sources {
```

Comment:
I don't think this is right. We don't want to mix ruleData across different sources. This should probably be done from `fetchPolicySources` so each evaluator only sees the sources relevant to them.

Reply:
That was my thought exactly, but `Evaluate` runs with all sources, or did I miss this as well?

Reply:
The only place that actually processes `Sources` is `NewApplicationSnapshotImage`. That iterates of the source groups, and for each one, it calls `fetchPolicySources` and `newConftestEvaluator` (alias to `NewConftestEvaluator`). `ValidateImage` then iterates through each evaluator, and calls `Evaluate` on it.

So I think that piece is working as expected since each call to `NewConftestEvaluator` creates its own isolated work dir.

Reply:
It's confusing I think because `c.PolicySources` as used in Evaluate is a list of policy and data source urls for a single "sourceGroup", but the `spec.configuration.sources` is a list of "sourceGroups".